
rm(list=ls())
library(Seurat)
library(ggplot2)

setwd("/Users/bendulken/Dropbox/Aging Single Cell Paper/Final_Code/10xInputFiles")
load("all_exprs_matrix_agg_mod_tworeps.R")


svz <- new("seurat", raw.data = all_exprs_matrix_agg_mod)

#Establishes seurat matrix with cutoff of 300 genes per cell and >=1 cell expressing a gene
svz <- Setup(svz, min.cells = 1, min.genes = 300, do.logNormalize = T, total.expr = 1E4, project = "10X_SVZ_Young_Old_Perf")

#This section of code is QC that is taken directly from Seurat

#nGene and nUMI are automatically calculated for every object by Seurat. For non-UMI data, nUMI represents the sum of the non-normalized values within a cell
# We calculate the percentage of mitochondrial genes here and store it in percent.mito using the AddMetaData. The % of UMI mapping to MT-genes is a common scRNA-seq QC metric.
mito.genes <- grep("^MT-", rownames(svz@data), value = T)
percent.mito <- colSums(expm1(svz@data[mito.genes, ]))/colSums(expm1(svz@data))

#AddMetaData adds columns to object@data.info, and is a great place to stash QC stats
svz <- AddMetaData(svz, percent.mito, "percent.mito")
pdf("VlnPlot.pdf",height=5,width=10)
VlnPlot(svz, c("nGene", "nUMI", "percent.mito"), nCol = 3)
dev.off()

par(mfrow = c(1, 2))
GenePlot(svz, "nUMI", "percent.mito")
GenePlot(svz, "nUMI", "nGene")


svz <- MeanVarPlot(svz ,fxn.x = expMean, fxn.y = logVarDivMean, x.low.cutoff = 0.0125, x.high.cutoff = 3, y.cutoff = 0.5, do.contour = F)
svz <- PCA(svz, pc.genes = svz@var.genes, do.print = TRUE, pcs.print = 5, genes.print = 5)

PCHeatmap(svz, pc.use = 1, cells.use = 100, do.balanced = TRUE)
PCElbowPlot(svz)


svz <- RunTSNE(svz, dims.use = 1:10, do.fast = T)
load("/Volumes/LaCie/05222017_10xGenomics_plusperfusion_v2/Analysis_twoperfrep/svz_tsne_young_old_tworeps.Robj")

pdf("tsne_young_old.pdf",height=6,width=6)
TSNEPlot(svz,do.return=T,do.label=F,no.legend=F,colors.use=c("darkorange","firebrick","deepskyblue","slateblue"))
dev.off()


fac<-c()
for(i in 1:length(rownames(svz@tsne.rot))){
  age<-strsplit(rownames(svz@tsne.rot)[i],"[_]")[[1]][1]
  fac<-c(fac,age)
}
fac<-factor(fac, levels=c("Young1","Young2","Old1","Old2"), ordered=T)

#Manually plot TSNE for Fig 1B, independently regenerated by Matt
data<-data.frame(svz@tsne.rot,fac=fac)
p<-ggplot(data)
p<-p+geom_point(aes(x=data[,1],y=data[,2],color=data$fac),size=1,alpha=0.5)
p<-p+labs(x="TSNE Coordinate 1", y="TSNE Coordinate 2")
p<-p+scale_colour_manual(values=c("deepskyblue","slateblue","darkorange","firebrick"))
pdf("tsne_young_old_manualplot.pdf",height=5,width=6)
print(p)
dev.off()

#Manual Plot with NSCs highlighted
cellidentities<-as.vector(read.table("cellidentities.txt")[,1])
fac2<-as.vector(fac)
fac2[cellidentities=="Endothelial"|cellidentities=="Microglia"|cellidentities=="Oligodendrocytes"|cellidentities=="Neuroblasts"|cellidentities=="Pericytes"|cellidentities=="T_Cells"|cellidentities=="OPC"]<-"notinteresting"
fac2<-factor(fac2, levels=c("notinteresting","Young1","Young2","Old1","Old2"), ordered=T)
data<-data.frame(svz@tsne.rot,fac=fac2)
p<-ggplot(data)
p<-p+geom_point(aes(x=data[,1],y=data[,2],color=data$fac),size=1,alpha=0.7)
p<-p+labs(x="TSNE Coordinate 1", y="TSNE Coordinate 2")
p<-p+scale_colour_manual(values=c("slategrey","deepskyblue","slateblue","darkorange","firebrick"))
pdf("tsne_young_old_NSCs_highlighted_manualplot.pdf",height=5,width=6.5)
print(p)
dev.off()

#Saves Seurat object for future calls, prior to clustering cell types
save(svz, file = "/Volumes/LaCie/05222017_10xGenomics_plusperfusion_v2/Analysis_twoperfrep/svz_tsne_young_old_tworeps.Robj")

#Calls significant clusters using the first 10 principle components
svz <- FindClusters(svz, pc.use = 1:10, resolution = 0.2, print.output = 0, save.SNN = T)

#Saves Seurat object for future calls after clustering cell types
save(svz, file = "/Volumes/LaCie/05222017_10xGenomics_plusperfusion_v2/Analysis_twoperfrep/svz_tsne_young_old_tworeps_clusters.Robj")


svz.markers <- FindAllMarkers(svz, only.pos = TRUE, min.pct = 0.25, thresh.use = 0.25)

write.table(svz.markers,"svz_cluster_markers_tworeps.txt")

svz.markers %>% group_by(cluster) %>% top_n(5, avg_diff) -> top10
# setting slim.col.label to TRUE will print just the cluster IDS instead of every cell name

#Marker Heatmap for Extended Data Fig 1A.
pdf("heatmap_tworeps.pdf",height=10,width=20)
DoHeatmap(svz, genes.use = top10$gene, order.by.ident = TRUE, slim.col.label = T, remove.key = TRUE)
dev.off()

jpeg("heatmap_tworeps.jpeg",width=400,height=1200,quality=100)
DoHeatmap(svz, genes.use = top10$gene, order.by.ident = TRUE, slim.col.label = F, remove.key = F)
dev.off()
# 
#
save(svz, file = "svz_tsne_young_old_tworeps_celllabeled_clusters_markers.Robj")
save(top10, file = "top10markers.Robj")


